### PR TITLE
Fix Numba array backend edge cases and maximize pure Python test coverage

### PR DESCRIPTION
**Pull request by player3zxcvbnm**
Hi,
I originally started looking into src/arviz_stats/numba/array.py just to bump up the test coverage, but I ended up stumbling onto a few edge-case bugs triggered by unhandled input types.
Most of the changes here are just handling None values properly (like testing axis=None in kde or bins=None in histogram). 
I noticed that all the remaining uncovered lines are strictly inside the guvectorize and jit blocks. Is it normal that pytest-cov can't see inside these compiled Numba functions, or is there a special way you usually test them?
Would really appreciate any feedback.

### CODE REVIEW

File: tests/numba/test_array.py [hunk 1]
```diff
@@ -139,26 +202,25 @@ def test_kde_basic(self, rng):
...
-    @pytest.mark.parametrize("axis", [0, 1, -1, -2])
-    def test_kde_axis(self, rng, axis):
+    @pytest.mark.parametrize(
+        "axis,expected_shape,expected_bw_shape",
+        [
+            (0, (20, 30, 128), (20, 30)),
+            (1, (10, 30, 128), (10, 30)),
+            (-1, (10, 20, 128), (10, 20)),
+            (-2, (10, 30, 128), (10, 30)),
+            (None, (128,), ()),
```
[the remark this answers is not in the document]
Comment (player3zxcvbnm) [reply tests/numba/test_array.py]: Thanks again for the review! Will   
take a look at it and push the updates soon.

**Review by OriolAbril — COMMENTED** [review]
This looks good and we'll be able to merge once CI is all green. At the very least you'll need to run `tox -e check`, you might also need to rebase on `main`.

File: src/arviz_stats/numba/array.py [hunk 2]
```diff
@@ -66,15 +66,22 @@ def quantile(self, ary, quantile, axis=-1, method="linear", skipna=False, weight
         axes = axis
         if axes is not None:
             ary, axes = process_ary_axes(ary, axes)
-            axes = [(-1,), (0,), (0,)]
+            axes = [axes, (0,), (0,)]
```
Comment (OriolAbril) [inline src/arviz_stats/numba/array.py]: This needs to be undone. I confused the process_ary_axes in the base module with this one. Here the axis we need to act on are moved to the end of the array and stacked together. We should go back to `(-1,)` and I would also update the `process_ary_axes` to not return the 2nd output because we don't use it anywhere and is only a source of confusion.

**Review by OriolAbril — COMMENTED** [review]
As explained in https://python.arviz.org/projects/stats/en/latest/contributing/testing.html, the minimal and xarray testing jobs do not install numba, so the full job is the most relevant for this PR. You can see the failing tests at the bottom of the logs: https://github.com/arviz-devs/arviz-stats/actions/runs/24003327906/job/70004650837?pr=320#step:5:82839. In this case, the failures might be related to the incorrect suggestion I made about the `-1`/`axes` in axis-axes processing

File: src/arviz_stats/numba/array.py (hunk 2 above)
Comment (player3zxcvbnm) [reply src/arviz_stats/numba/array.py]: Update: I realized that in my previous push I missed the tests in `test_array.py` and a few internal call sites (like in `kde`) that were still expecting the second return value.

I've just force-pushed a final clean version that fixes all those errors and verified it locally with `tox -r`.

**Review by OriolAbril — APPROVED** [review]
